src/main/java/picard/util/IntervalList/IntervalListScattererByIntervalCountWithDistributedRemainder.java

```diff
@@ -0,0 +1,24 @@
+package picard.util.IntervalList;
+
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Scatters {@link IntervalList} by into `interval count` shards so that resulting {@link IntervalList}'s have
+ * approximately same number of intervals in them. The "remainder" intervals are distributed over the last lists.
+ */
+public class IntervalListScattererByIntervalCountWithDistributedRemainder extends IntervalListScattererByIntervalCount {
+
+    @Override
+    public List<Interval> takeSome(final Interval interval, final long idealSplitWeight, final long currentSize, final double projectSizeOfRemaining) {
+        final double amount = projectSizeOfRemaining - currentSize;
```

Comment:
I know you're just copying this structure from `IntervalListScattererByIntervalCount`, but this feels like a very roundabout way to write `if (projectSizeOfRemaining > currentSize)`